internal/weapons/sword/haran/haran.go

```diff
@@ -34,33 +35,36 @@ func weapon(char core.Character, c *core.Core, r int, param map[string]int) stri
 		Expiry: -1,
 	})
 
-	c.Events.Subscribe(core.OnAttackWillLand, func(args ...interface{}) bool {
+	c.Events.Subscribe(core.PostSkill, func(args ...interface{}) bool {
 		if c.ActiveChar == char.CharIndex() {
 			return false
 		}
-		if c.F > wavespikeICD && args[1].(*core.AttackEvent).Info.AttackTag == core.AttackTagNormal {
+		if c.F > wavespikeICD {
 			wavespikeStacks++
 			if wavespikeStacks > maxWavespikeStacks {
 				wavespikeStacks = maxWavespikeStacks
 			}
+			c.Log.NewEvent("Haran gained a wavespike stack", core.LogWeaponEvent, char.CharIndex(), "stack", wavespikeStacks)
 			wavespikeICD = c.F + 0.3*60
 		}
 
 		return false
 	}, fmt.Sprintf("wavespike-%v", char.Name()))
+
 	val := make([]float64, core.EndStatType)
 	c.Events.Subscribe(core.PostSkill, func(args ...interface{}) bool {
 		if c.ActiveChar != char.CharIndex() {
 			return false
 		}
+		stackSnapshot = wavespikeStacks
 		char.AddPreDamageMod(core.PreDamageMod{
 			Key:    "ripping-upheaval",
 			Expiry: c.F + 60*8,
 			Amount: func(atk *core.AttackEvent, t core.Target) ([]float64, bool) {
 				if atk.Info.AttackTag != core.AttackTagNormal {
 					return nil, false
 				}
-				val[core.DmgP] = (0.2 * float64(r) * 0.04) * float64(wavespikeStacks)
+				val[core.DmgP] = (0.15 + float64(r)*0.05) * float64(stackSnapshot)
```

Comment:
this line should be after L59 (actually there's no reason for `stackSnapshot` variable). no point redoing the calculation every single time the functiong ets called